### PR TITLE
[17064] Add ROS 2 CLI module to Fast DDS Discovery Server Tutorial

### PR DESCRIPTION
**Pull request by MRicoIE2CS**
Update in Fast DDS Discovery Server Tutorial for ROS 2.

New paragraph: ROS 2 Introspection.
It explains how to use ROS 2 Introspection with ROS 2 CLI and ROS 2 Daemon within the Fast DDS Discovery Server architecture, using Super Client capabilities.

This PR is a manual merge between this ros2/ros2_documentation#1028 and eProsima/Fast-DDS-docs#242.

Signed-off-by: jparisu <javierparis@eprosima.com>
Signed-off-by: Mikel Rico <mikelrico@eprosima.com>

### CODE REVIEW

File: source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst [hunk 1]
```diff
@@ -51,7 +51,7 @@ The following figure shows the decrease in discovery messages:
     :align: center
 
 This architecture reduces the number of messages sent between the server and clients dramatically.
-In the following graph, the reduction in network traffic over the discovery phase for the `RMF Clinic demonstration <https://github.com/osrf/rmf_demos>`__ is shown:
+In the following graph, the reduction in network traffic over the discovery phase for the `RMF Clinic demonstration <https://github.com/osrf/rmf_demos#Clinic-World>`__ is shown:
```
Comment (EduPonz) [inline source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst]:
```suggestion
In the following graph, the reduction in network traffic over the discovery phase for the `RMF Clinic demonstration <https://github.com/open-rmf/rmf_demos#Clinic-World>`__ is shown:
```

File: source/Tutorials/Advanced/Discovery-Server/scripts/super_client_configuration_file.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<dds>
+    <profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
+        <participant profile_name="super_client_profile" is_default_profile="true">
+            <rtps>
+                <prefix>44.49.53.43.53.45.52.56.45.52.5f.30</prefix>
```
Comment (EduPonz) [inline source/Tutorials/Advanced/Discovery-Server/scripts/super_client_configuration_file.xml]:
```suggestion
```

File: source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst [hunk 3]
```diff
@@ -285,10 +285,125 @@ We should see how ``Listener 1`` is receiving messages from both talker nodes, w
...
+This will allow the Daemon to discover the entire Node graph, and to receive every topic and endpoint information.
+To do so, a Fast DDS XML configuration file is used to configure the ROS 2 Daemon and CLI tools.
+
+.. warning::
+    Although it is possible to run the ROS 2 Daemon as a **Server**, this is not recommended since the daemon will stop
+    after two hours of inactivity, taking the **Server** down with it.
+
+Below you can find a XML configuration file which will configure every new participant as a **Super Client**.
+
+* :download:`XML Super Client configuration file <scripts/super_client_configuration_file.xml>`
```
Comment (EduPonz) [inline source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst]: I think I rather see an snippet here using a literal include

File: source/Tutorials/Advanced/Discovery-Server/scripts/super_client_configuration_file.xml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+                <builtin>
+                    <discovery_config>
+                        <discoveryProtocol>SUPER_CLIENT</discoveryProtocol>
+                        <discoveryServersList>
+                            <RemoteServer prefix="44.53.00.5f.45.50.52.4f.53.49.4d.41">
```
Comment (EduPonz) [inline source/Tutorials/Advanced/Discovery-Server/scripts/super_client_configuration_file.xml]: I'd add a comment here stating that the prefix should be updated according to the server ID passed on the CLI. We somehow need a note which links to Fast DDS docs where we explain this

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst (hunk 1 above)
Comment (MRicoIE2CS) [reply source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst]: Done

File: source/Tutorials/Advanced/Discovery-Server/scripts/super_client_configuration_file.xml (hunk 2 above)
Comment (MRicoIE2CS) [reply source/Tutorials/Advanced/Discovery-Server/scripts/super_client_configuration_file.xml]: Done

File: source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst (hunk 3 above)
Comment (MRicoIE2CS) [reply source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst]: Done

File: source/Tutorials/Advanced/Discovery-Server/scripts/super_client_configuration_file.xml (hunk 4 above)
Comment (MRicoIE2CS) [reply source/Tutorials/Advanced/Discovery-Server/scripts/super_client_configuration_file.xml]: Done

File: source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst [hunk 5]
```diff
@@ -322,11 +322,45 @@ To do so, a Fast DDS XML configuration file is used to configure the ROS 2 Daemo
     Although it is possible to run the ROS 2 Daemon as a **Server**, this is not recommended since the daemon will stop
     after two hours of inactivity, taking the **Server** down with it.
 
-Below you can find a XML configuration file which will configure every new participant as a **Super Client**.
+Below you can find a XML configuration profile, which for this tutorial, should be saved in the working directory as ```super_client_configuration_file.xml``` file.
```
Comment (EduPonz) [inline source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst]:
```suggestion
Below you can find a XML configuration profile, which for this tutorial should be saved in the working directory as ```super_client_configuration_file.xml``` file.
```

File: source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst [hunk 6]
```diff
@@ -322,11 +322,45 @@ To do so, a Fast DDS XML configuration file is used to configure the ROS 2 Daemo
...
+            </participant>
+        </profiles>
+    </dds>
 
-* :download:`XML Super Client configuration file <scripts/super_client_configuration_file.xml>`
 
-First of all, instantiate a Discovery Server using `Fast DDS CLI <https://fast-dds.docs.eprosima.com/en/latest/fastddscli/cli/cli.html#discovery>`__
+.. note::
+
+    Under the *RemoteServer* tag, for *prefix* attribute, its value should be updated according to the server ID passed on the CLI, following this guide of the `Fast DDS CLI <https://fast-dds.docs.eprosima.com/en/latest/fastddscli/cli/cli.html#discovery>`__.
```
Comment (EduPonz) [inline source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst]:
```suggestion
    Under the *RemoteServer* tag, the *prefix* attribute value should be updated according to the server ID passed on the CLI (see `Fast DDS CLI <https://fast-dds.docs.eprosima.com/en/latest/fastddscli/cli/cli.html#discovery>`__).
```

**Review by EduPonz — CHANGES_REQUESTED** [review]
NITs

File: source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst (hunk 5 above)
Comment (MRicoIE2CS) [reply source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst]: Done

File: source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst (hunk 6 above)
Comment (MRicoIE2CS) [reply source/Tutorials/Advanced/Discovery-Server/Discovery-Server.rst]: Done

**Review by EduPonz — APPROVED** [review]

**Pull request merged**